pyproject.toml

```diff
@@ -1,32 +1,109 @@
+requires = ["hatchling>=1.0", "jupyterlab>=4.0.0a25,<5"]
+build-backend = "hatchling.build"
 
-[license]
-file="LICENSE"
+[project]
+name = "notebook"
+description = "Jupyter Notebook - A web-based notebook environment for interactive computing"
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.7"
+authors = [
+    { name = "Jupyter Development Team", email = "jupyter@googlegroups.com" },
+]
+keywords = [
+    "Jupyter",
+    "JupyterLab",
+    "Notebook",
+]
+classifiers = [
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "jupyter_server>=1.16.0,<2",
+    "jupyterlab>=4.0.0a24,<5",
+    "jupyterlab_server>=2.13,<3",
+    "notebook_shim>=0.1,<0.2",
+    "tornado>=6.1.0",
+]
+dynamic = ["version"]
 
-[tool.jupyter-packaging.options]
-skip-if-exists = ["notebook/labextension/static/style.js", "notebook/static/bundle.js"]
-ensured-targets = ["notebook/labextension/static/style.js", "notebook/static/bundle.js"]
+[project.scripts]
+jupyter-notebook = "notebook.app:main"
 
-[tool.jupyter-packaging.builder]
-factory = "jupyter_packaging.npm_builder"
+[project.urls]
+Documentation = "https://jupyter-notebook.readthedocs.io/"
+Homepage = "https://github.com/jupyter/notebook"
+Source = "https://github.com/jupyter/notebook"
+Tracker = "https://github.com/jupyter/notebook/issues"
 
-[tool.jupyter-packaging.build-args]
-build_cmd = "build:prod"
-npm = ["jlpm"]
+[project.optional-dependencies]
+test = [
+    "coverage",
+    "nbval",
+    "pytest>=6.0",
+    "pytest-cov",
+    "requests",
+    "pytest-tornasync",
+    "pytest-timeout",
+    "pytest-console-scripts",
+    "ipykernel",
+    "jupyterlab_server[test]>=2.13,<3",
+]
+dev = [
+    "pre-commit",
+    "bump2version",
+    "hatchling"
+]
+
+[tool.hatch.version]
+path = "notebook/_version.py"
+source = "code"
 
-[tool.check-manifest]
-ignore = ["app/**", "binder/**", "buildutils/**", "docs/**", "packages/**", "codecov.yml", "*.json", "yarn.lock", "readthedocs.yml", ".bumpversion.cfg", ".*", "lint-staged.config.js", "*.svg", "notebook/labextension/**", "notebook/schemas/**", "notebook/static/**", "notebook/template/**", "ui-tests/**"]
+[tool.hatch.build.targets.wheel.shared-data]
+"notebook/labextension" = "share/jupyter/labextensions/@jupyter-notebook/lab-extension"
+"notebook/schemas/@jupyter-notebook" = "share/jupyter/lab/schemas/@jupyter-notebook"
+"jupyter-config/jupyter_server_config.d" = "etc/jupyter/jupyter_server_config.d"
+
+
+[tool.hatch.build]
+artifacts = [
+    "notebook/labextension",
+    "notebook/static",
+    "notebook/schemas"
+]
```

Comment:
Do you want these in everything or just the sdist?

Reply:
Previously the files were in both so I kept it that way.  I agree it doesn't make much sense to have them in the wheel since the files are only used in the shared-data directory.  What do you think @jtpio?

Reply:
For me, putting this under `[tool.hatch.build.targets.sdist]` instead brings the wheel size from ~3 MiB (see https://pypi.org/project/notebook/7.0.0a4/#files) down to ~50 KiB

Reply:
Actually, `"notebook/static"` is definitely needed, it is not used as `shared-data`, but the static assets used for the `notebook` app-specific page content.  I *think* the other two folders can be excluded, but @jtpio would know beter.

Reply:
`notebook/static` is indeed needed.

And it looks like `notebook/labextension` and `notebook/schemas` can be dropped if they are already specified in `tool.hatch.build.targets.wheel.shared-data`?